### ISSUE TITLE
Pvnet 2 and naitontal_xg stopped working

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/nwp-consumer#24]
- Pvnet 2 stopped working yesterday 22.45 at and 
- National_xg stoped on 22nd around 18.00
- Pvnet 1 stopped working on 22nd around 14.00

National_xg loads the `latest.zarr` from data-national but the last init time is `2023-08-22T12:00:00`. So perhaps the save .zarr method isnt working?

**Comment by peterdudfield** [comment]
I'm just going to re trigger the old one, so there are some forecasts. This helps me with fixing a different bug

**Comment by peterdudfield** [comment]
Shall i bump memory up to 5GB and see if it works?

**Comment by peterdudfield** [comment]
> Shall i bump memory up to 5GB and see if it works?

or could you @devsjc ?

**Comment by devsjc** [comment]
Can do! I didn't think this runing for national yet, or did you modify the national consumer as well to use this image?

**Comment by devsjc** [comment]
I can see that you have. I'll make the change as well but we neet to set the raw dir and zarr dir differently for the case this is running as the national consumer - they don't go in `raw` and `data` but rather `raw-national` and `data-national`.

**Comment by devsjc** [comment]
PR to modify the command for national. and increase the memory: https://github.com/openclimatefix/ocf-infrastructure/pull/288

**Comment by devsjc** [comment]
Terraform run implementing the above [here](https://app.terraform.io/app/openclimatefix/workspaces/nowcasting_infrastructure_development-eu-west-1/runs/run-uQPv9LE6f6mGYmfi).

**Comment by devsjc** [comment]
The national forecast is now failing with 

```
assert ukv.y_osgb[0] > ukv.y_osgb[1], "UKV must run from top-to-bottom.
```

see [Cloudwatch logs](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#logsV2:log-groups/log-group/%2Faws%2Fecs%2Fforecast_pvnet%2F/log-events/streaming%2Fforecast%2F690393e76ee34cbd959c8a178a56362b).

I'll update this along with the variable reordering.

**Comment by peterdudfield** [comment]
Thanks for all this - 

<img width="807" alt="Screenshot 2023-08-24 at 12 16 07" src="https://github.com/openclimatefix/nwp-consumer/assets/34686298/c6af1258-95a9-4a56-8c14-768f5512a0fb">
the re-ordering should solve this

- National_xg might need some work on the forecast side. It does some checks 

<img width="1108" alt="Screenshot 2023-08-24 at 12 17 09" src="https://github.com/openclimatefix/nwp-consumer/assets/34686298/6fe59c6e-fd82-4206-8e68-4bcc7aee8123">

**Comment by devsjc** [comment]
Pull request for the above here https://github.com/openclimatefix/nwp-consumer/pull/25

**Comment by devsjc** [comment]
Terraform run updating to version incorporating the above pull request [here](https://app.terraform.io/app/openclimatefix/workspaces/nowcasting_infrastructure_development-eu-west-1/runs/run-VQtgK49z1As7VTyT)

**Comment by peterdudfield** [comment]
Current status
- pvnet 1: failling not having data at center Example is wrong shape! x_center_osgb=274767.28508405434, y_center_osgb=61482.03366374501
- pvnet 2: cant get data for an example 

<img width="717" alt="Screenshot 2023-08-24 at 14 19 12" src="https://github.com/openclimatefix/nwp-consumer/assets/34686298/6e1f44cf-ecac-4221-96aa-91795aee5e12">

- forecast-national: size is different, this is used when making the sea mask. This could be solved by re running something in this repo 

<img width="1221" alt="Screenshot 2023-08-24 at 14 20 16" src="https://github.com/openclimatefix/nwp-consumer/assets/34686298/d03a0ca6-aa87-4c9f-ba4c-af0414ea7072">

**Comment by peterdudfield** [comment]
I think I see the error, the x,y in the dataset are ints, where as x,y in the old dataset are osgb numbers. I totally understand how this is not clear though

**Comment by peterdudfield** [comment]
It looks like PVnet use lat lon/lat. Perhaps we just need to look at the area covered. 

New dataset goes from lat = (49.43289499675095, 61.18842789168269) and lon = (-13.10798670287675, 3.6463370654506146)

**Comment by devsjc** [comment]
That lines up with what is in the grib files, using `grib_dump`:

```
(grib2/tables/29/3.2.table)  
  shapeOfTheEarth = 4;
  numberOfPointsAlongXAxis = 455;
  numberOfPointsAlongYAxis = 639;
  latitudeOfFirstGridPointInDegrees = 49.4329;
  longitudeOfFirstGridPointInDegrees = -10.451;
  standardParallelInMicrodegrees = 54900000;
  standardParallelInDegrees = 54.9;
  centralLongitudeInMicrodegrees = -2500000;
  centralLongitudeInDegrees = -2.5;
  # (3=1)  i direction increments given ;(4=1)  j direction increments given ;(5=0)  Resolved u- and v- components of vector quantities relative to easterly and northerly directions :grib2/tables/29/3.3.table 
  # flags: 00110000
  resolutionAndComponentFlags = 48;
  xDirectionGridLengthInMillimetres = 2000000;
  xDirectionGridLengthInMetres = 2000;
  yDirectionGridLengthInMillimetres = 2000000;
  yDirectionGridLengthInMetres = 2000;
  iScansNegatively = 0;
  jScansPositively = 1;
  jPointsAreConsecutive = 0;
  alternativeRowScanning = 0;
  gridType = lambert_azimuthal_equal_area;
  NV = 0;
  # Analysis or forecast at a horizontal level or in a horizontal layer at a point in time (grib2/tables/29/4.0.table)  
  productDefinitionTemplateNumber = 0;
```

So the new dataset does seem to be mapping those values through correctly.

**Comment by devsjc** [comment]
After updating to the version created in the above pull request, we are at the following point:

`forecast` still fails with a wrong shape error, see [full cloudwatch logs](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#logsV2:log-groups/log-group/%2Faws%2Fecs%2Fforecast%2F/log-events/streaming%2Fforecast%2F08b9869730324862b09ef0978c59d518):

```
expected shape=(1, 4, 64, 64)
actual shape (1, 4, 63, 64)
```

So close!

And `forecast-national` fails with an assertion error in `/app/gradboost_pv/inference/models.py`, see [full logs in cloudwatch](https://eu-west-1.console.aws.amazon.com/ecs/v2/clusters/Nowcasting-development/tasks/34ecf586a69b4ecb840dda0e2e3b7558/logs?region=eu-west-1):

```
  assert set(self._config.nwp_variables).issubset(
            data.nwp.coords[self._config.nwp_variable_name].values
```

So far as I can tell that `nwp_variable_name` is set as "channel" [here](https://github.com/openclimatefix/uk-pv-national-xg/blob/main/gradboost_pv/inference/models.py#L143C31-L143C38) which wasn't ever a variable as far as I'm aware...

**Comment by peterdudfield** [comment]
Pvnet 2 is working
Pvnet 1 should be solved by - https://github.com/openclimatefix/nwp-consumer/pull/28

**Comment by peterdudfield** [comment]
site forecasts aslo has an error
https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#logsV2:log-groups/log-group/$252Faws$252Fecs$252Fpvsite_forecast$252F/log-events/streaming$252Fforecast$252F432fff7c82b140638be0158a0ca8b8ea

si10 not in nwp data from s3://arn:aws:s3:::nowcasting-nwp-development/data/latest.zarr

**Comment by peterdudfield** [comment]
Need to make sure `nwp-national` and `nwp` save to their correct data folder. I think nwp-national is being saved to `3://arn:aws:s3:::nowcasting-nwp-development/data/latest.zarr` and it should be saved to `s3://arn:aws:s3:::nowcasting-nwp-development/data-national/latest.zarr`

**Comment by peterdudfield** [comment]
should make x and y to its https://github.com/openclimatefix/nwp-consumer/pull/28
https://github.com/openclimatefix/nwp-consumer/pull/29 should fix wind variables

**Comment by peterdudfield** [comment]
Currentl status
- national-forecast - needs fix on mask
- pvent 1 - small bug, might be fixed by rolling back to 1.3.22
- pvetn 2 - wokring
- site forecast - should be fixed by #29
